modules/openapi-generator/src/test/java/org/openapitools/codegen/java/AbstractJavaCodegenTest.java

```diff
@@ -572,7 +568,9 @@ public void toDefaultValueDateTimeLegacyTest() {
         defaultValue = codegen.toDefaultValue(dateSchema);
 
         // dateLibrary <> java8
-        Assert.assertNull(defaultValue);
+        if (defaultValue.contains("HKT")) {
```

Comment:
:x: This will only work when the local `ZoneId.systemDefault()` happens to be `HKT` which I imagine is not the case for every developer. I'd suggest to test with a named `ZoneId`, so that it is portable.

❗ Furthermore, I'd try to avoid conditionals in tests as much as possible. This condition can be false when it should actually be true and nothing would be failing.